### PR TITLE
ForbiddenVoid: New option 'ignoreUsageInGenerics'

### LINKED ISSUE #1724: ForbiddenVoid and TestObserver<Void>
This is a follow-up on #1472 - I just stumbled upon the very same issue with RxJava and TestObserver of a Completable:

```kotlin
fun testMe(): TestObserver<Void> = Completable.complete().test()
```

is then also flagged with ForbiddenVoid. IMHO this rule should get some configuration, such as onlyCheckTypesInPackage which we would naturally configure to our project package path and the rule could then ignore types that reside in different (base) packages from that.

### PR DESCRIPTION
**Pull request by realdadfish**
Fixes #1724

### CODE REVIEW

**Comment by schalkms** [comment]
Please execute `gradle build` locally in order to generate the documentation.
Detekt's documentation is automatically generated from the comments in the rule class.
This helps us to have an up to date documentation.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt [hunk 1]
```diff
@@ -69,8 +69,7 @@ class ForbiddenVoidSpec : Spek({
 
                     class B : A() {
                         override fun method(param: Foo<Void>) : Foo<Void> {
-                            @Suppress("ForbiddenVoid")
-                            return Foo<Void>()
+                            throw IllegalStateException()
```
Comment (schalkms) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt]: Why did you change this?

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt [hunk 2]
```diff
@@ -107,5 +106,42 @@ class ForbiddenVoidSpec : Spek({
                 assertThat(findings).hasSize(2)
             }
         }
+        describe("ignoreUsageInGenerics is enabled") {
```
Comment (schalkms) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt]: I would prefer to have the following test cases:

1. nested generic void usage (e.g. `A<B<Void>>`)
2. Dictionary (e.g. `Map<Int, Void>`)

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt (hunk 1 above)
Comment (realdadfish) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt]: It served no purpose for the particular test, but rather confused because of the additional `@Suppress("ForbiddenVoid")`.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt (hunk 2 above)
Comment (realdadfish) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt]: Will add these.
